sentry/src/routes/campaign.rs

```diff
@@ -1,56 +1,286 @@
+        // !WARNING!: totalSpent != sum(campaign.map(c => c.spending)) therefore we must always calculate remaining funds based on total_deposit - lastApprovedNewState.spenders[user]
+        // *NOTE*: To close a campaign set campaignBudget to campaignSpent so that spendable == 0
+
+        if let Some(new_budget) = modified_campaign.budget {
+            let old_remaining = get_remaining_for_campaign(&redis, campaign.id).await?.ok_or(Error::FailedUpdate("No remaining entry for campaign".to_string()))?;
+
+            let campaign_spent = campaign.budget.checked_sub(&old_remaining).ok_or(Error::Calculation)?;
+            if campaign_spent >= new_budget {
+                return Err(Error::NewBudget("New budget should be greater than the spent amount".to_string()));
+            }
+
+            // Separate variable for clarity
+            let old_budget = campaign.budget;
+
+            match new_budget.cmp(&old_budget) {
+                Ordering::Greater | Ordering::Equal => {
+                    let new_remaining = old_remaining.checked_add(&new_budget.checked_sub(&old_budget).ok_or(Error::Calculation)?).ok_or(Error::Calculation)?;
+                    let amount_to_incr = new_remaining.checked_sub(&old_remaining).ok_or(Error::Calculation)?;
+                    increase_remaining_for_campaign(&redis, campaign.id, amount_to_incr).await?;
+                },
+                Ordering::Less => {
+                    let new_remaining = old_remaining.checked_add(&old_budget.checked_sub(&new_budget).ok_or(Error::Calculation)?).ok_or(Error::Calculation)?;
+                    let amount_to_decr = new_remaining.checked_sub(&old_remaining).ok_or(Error::Calculation)?;
+                    let decreased_remaining = decrease_remaining_for_campaign(&redis, campaign.id, amount_to_decr).await.ok_or(Error::FailedUpdate("Could't decrease remaining".to_string()))?;
+                    // If it goes below 0 it will still return 0
+                    if decreased_remaining.eq(&UnifiedNum::from_u64(0)) {
+                        return Err(Error::NewBudget("No budget remaining after decreasing".to_string()));
+                    }
+                }
+            }
+        };
+
+        let accounting_spent = get_accounting_spent(pool.clone(), &campaign.creator, &campaign.channel.id()).await?;
+
+        let latest_spendable = fetch_spendable(pool.clone(), &campaign.creator, &campaign.channel.id()).await?;
+
+        // Gets the latest Spendable for this (spender, channelId) pair
+        let total_deposited = latest_spendable.deposit.total;
+
+        let total_remaining = total_deposited.checked_sub(&accounting_spent).ok_or(Error::Calculation)?;
+        let campaigns_for_channel = get_campaigns_by_channel(&pool, &campaign.channel.id()).await?;
+        // campaign.budget should 100% exist, therefore it should be safe to just unwrap?
+        let current_campaign_budget = modified_campaign.budget.ok_or(campaign.budget).unwrap();
```

Comment:
it's better to use `unwrap_or()` in this case. It will use the `campaign.budget` and will directly return to you `UnifiedNum` instead of `Option<UnifiedNum>`

- [ ] Use `unwrap_or()`